src/main.mo

```diff
@@ -119,13 +121,27 @@ shared ({ caller = init_minter }) actor class Canister(cid : Principal) = myCani
 
   func _setTimers() {
     Timer.cancelTimer(_timerId);
+    Timer.cancelTimer(_revealTimerId);
 
     _timerId := Timer.recurringTimer(Env.timersInterval, func(): async () {
       ignore cronSettlements();
       ignore cronDisbursements();
       ignore cronSalesSettlements();
       ignore cronFailedSales();
     });
+
+    if (Env.delayedReveal) {
+      let revealTime = Env.publicSaleStart + Env.revealDelay;
+      let delay = Int.abs(Int.max(0, revealTime - Time.now()));
```

Comment:
what if delay is `0`, because we upgrade the canister after the delay has already happend? seems like in this case we still setup a timer, even though it shouldn't have an effect because of the  `_isShuffled` stable variable, i think we can instead just not set a timer at all.

Reply:
Added check if already shuffled